addon/index.js

```diff
@@ -25,9 +26,17 @@ export default Mixin.create({
 
 		if (get(this, 'isFastBoot')) { return; }
 
-    this.get('scheduler').scheduleWork('afterContentPaint', () => {
-      this.updateScrollPosition(transitions);
-    });
+    const delayScrollTop = get(this, 'service.delayScrollTop');
+
+    if (!delayScrollTop) {
+      scheduleOnce('render', this, () => this.updateScrollPosition(transitions));
```

Comment:
The reason why this works is b/c by this time the callback runs, the `pushState` function has already ran, thus we can update the scroll position.

Reply:
We could also do one/two better and do this after `sync` (which is where HistoryLocation updates are batched _I think_), which would be `actions` or `routerTransitions`, but `render` seems safe enough.